### PR TITLE
Add scispacy tokenisation

### LINKED ISSUE #15: Refine cutting text into 3 sentences
It doesn't happen very often, but sometimes there are no fullstops in the text, so it doesn't split neatly into sentences.
Also if there are fullstops in sentences such as "e.g." this doesnt work very well either.

Think of a better way at some point. @nsorros says " in the future we could experiment with scispacy that should be better in splitting sentences on that domain."

e.g. 

 "paper_id": "0775644af19e13cb5a4214be4db3f8ad75acd2d4", 

snippet: " \u2022 A very recent study suggests that digestive symptoms such as loss of appetite and diarrhea are also common [8] \u2022 In contrast, upper airway catarrh syndrome, such as running nose or sneezing, are rare in COVID-19 patients [9] \u2022 According to the WHO, the three most common symptoms for COVID-19 are: fever, tiredness, and dry cough [10] \u2022 In addition to respiratory symptoms, some COVID-19 patients may exhibit GI symptoms [8] , suggesting unexplained diarrhea and may also be considered for screening \u2022 Recently, the American Academy of Otolaryngology -Head and Neck Surgery also recommended adding \"sudden loss of taste or smell\" as one of the symptoms for screening of COVID-19 [11] 4. How long does it usually take from infection to recovery? \u2022 The median time from symptom to recovery is 22 days in survivors and median duration of viral shedding is 20 days (longest 37 days) [12] \u2022 This information is useful for the strategic planning of cancer treatment during the COVID-19 outbreak: Delaying cancer treatment for COVD-19 positive cancer patients may be feasible for certain cancers, if expected delay is about 4 weeks \u2022 COVID-19 virus can spread via respiratory mucus or saliva droplets (coughing and talking), contact with bodily fluids (e.g"

### PR DESCRIPTION
**Pull request by nsorros**
Fixes #15

### CODE REVIEW

File: covid/question_covid.py [hunk 1]
```diff
@@ -67,10 +65,19 @@ def predict(self, question):
...
-            sentences = text.split('.')
-            n = 3
-            sentences_grouped = ['.'.join(sentences[i:i+n]) for i in range(0, len(sentences), n)]
-            for subtext in sentences_grouped:
+            sentences = [s.text for s in self.scispacy(text).sents]
+            
+            def yield_subtext(sentences):
+                subtext = ''
+                for i in range(len(sentences)):
+                    sent = sentences[i]
```
Comment (lizgzil) [inline covid/question_covid.py]: why not just `for sent in sentences:`

File: covid/question_covid.py [hunk 2]
```diff
@@ -67,10 +65,19 @@ def predict(self, question):
...
+                    if len(sent) + len(subtext) > 450:
+                        yield subtext
+                        subtext = sent
```
Comment (lizgzil) [inline covid/question_covid.py:75]: I don't think this works if `len(sent)>450`? 
e.g. 
`subtext =''`
`sent = 'kdfjhsdfkjhsdfjk ....skdjfhdkfjhdkfh'` (len>450)
`len(sent) + len(subtext) > 450` is true
`yield ''`
`subtext = sent`
the next iteration we will yield subtext which is `'kdfjhsdfkjhsdfjk ....skdjfhdkfjhdkfh'` (len>450)

I think in this (hopefully rare) situation we will just have to chop up subtext

**Review by lizgzil — APPROVED** [review]
I'm approving, but i think there is an edge case where this wont work as desired.
Although maybe its ok if a few sentences are over 450? and maybe it never happens anyway

File: covid/question_covid.py (hunk 2 above)
Comment (nsorros) [reply covid/question_covid.py:75]: there are some edge cases indeed, will try to fix today but merge first

File: covid/question_covid.py (hunk 1 above)
Comment (nsorros) [reply covid/question_covid.py]: no particular reason, i will change.

**Pull request merged**